web/installer

```diff
@@ -1366,15 +1417,15 @@ class HttpClient {
 
             // enabled request_fulluri unless it is explicitly disabled
             switch (parse_url($url, PHP_URL_SCHEME)) {
-                case 'http': // default request_fulluri to true
+                case 'http': // default request_fulluri to true for HTTP
                     $reqFullUriEnv = getenv('HTTP_PROXY_REQUEST_FULLURI');
                     if ($reqFullUriEnv === false || $reqFullUriEnv === '' || (strtolower($reqFullUriEnv) !== 'false' && (bool) $reqFullUriEnv)) {
                         $options['http']['request_fulluri'] = true;
                     }
                     break;
-                case 'https': // default request_fulluri to true
+                case 'https': // default request_fulluri to false for HTTPS
```

Comment:
This doesn't seem to coincide with the actual value being set though?

Reply:
I totally agree, but kept the code "as is", which does seem wrong. I'll fix this here. Will you fix it in Composer?

Reply:
@Seldaek do you know if that is a mistake in the code, or the comment?

Reply:
And it doesn't actually match the composer docs:

> If you use a proxy but it does not support the request_fulluri flag for HTTPS requests, then you should set this env var to false or 0 to prevent Composer from setting the request_fulluri option.

I took this from the Composer 2.0 branch (as suggested by @Seldaek) but I have changed this to what is currently on composer and getcomposer master.

Reply:
I asked you to take from 2.0 specifically because of this change, I think the 2.0 values are better defaults that reflect what others do by default too. request_fulluri defaults to false, so while the code sets it to true in the if, the comment says it's set to false by default because that is the default. I guess "left alone to its default of false" would be a more appropriate comment.

This potentially breaks stuff though so I didn't want to apply to 1.x, but IMO for the installer it's fine and it's a good way to test the code some more until 2.0 lands..

Reply:
Okay, thanks for the explanation. I'll rebase back to the original.